timedctl/timedctl.py

```diff
@@ -25,708 +24,568 @@
-    # select an activity
-    activity_obj = select_activity(date)
-    if timed.activities.delete(activity_obj[-1]):
-        msg(f"Activity {activity_obj[1]} deleted successfully.")
-        return
-    error_handler("ERR_ACTIVITY_DELETE_FAILED")
-
-
-@activity.command("generate-timesheet", aliases=["gts", "ts"])
-def activity_generate_timesheet():
-    """Generate the timesheet of the current activities."""
-    activities = timed.activities.get()
-    reports = timed.reports.get()
-    if activities:
-        for activity_obj in activities:
-            attr = activity_obj["attributes"]
-            if not attr["transferred"]:
-                # stop running activities
-                if not attr["to-time"]:
-                    attr["to-time"] = datetime.datetime.now()
-                # calculate duration
-                duration = attr["to-time"] - attr["from-time"]
-                # get task
-                task = activity_obj["relationships"]["task"]["data"]["id"]
-                # check if there is a report with the same comment that already exists
-                report = [
-                    x
-                    for x in reports
-                    if x["attributes"]["comment"] == attr["comment"]
-                    and x["relationships"]["task"]["data"]["id"] == task
-                ]
-                # if report has been found
-                if len(report) > 0:
-                    report = report[0]
-                    # deserialize the timedelta
-                    hours, minutes, seconds = report["attributes"]["duration"].split(
-                        ":",
-                    )
-                    old_duration = datetime.timedelta(
-                        hours=int(hours),
-                        minutes=int(minutes),
-                        seconds=int(seconds),
-                    )
-                    # calculate the new duration
-                    report["attributes"]["duration"] = old_duration + duration
-                    # update report
-                    timed.reports.patch(
-                        report["id"],
-                        report["attributes"],
-                        {"task": task},
-                    )
-                else:
-                    # create report
-                    res = timed.reports.post(
-                        {
-                            "duration": duration,
-                            "comment": attr["comment"],
-                        },
-                        {"task": task},
+
+        if res.status_code == requests.codes["created"]:
+            msg(f"Activity {comment} started successfully.")
+            return
+        # handle exception
+        error_handler("ERR_ACTIVITY_START_FAILED")
+
+    def stop_activity(self):
+        """Stop current activity."""
+        if not self.timed.activities.current:
+            error_handler("ERR_NO_CURRENT_ACTIVITY")
+        else:
+            self.timed.activities.stop()
```

Comment:
is the `else` really necessary?
```suggestion
        self.timed.activities.stop()
```